ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Types.hs

```diff
@@ -414,7 +423,14 @@ data PeerSelectionState peeraddr peerconn = PeerSelectionState {
        -- | 'PeerSelectionCounters' counters cache. Allows to only trace
        -- values when necessary.
        --
-       countersCache               :: !(Cache PeerSelectionCounters)
+       countersCache               :: !(Cache PeerSelectionCounters),
+
+       -- | Cached value of 'LedgerStateJudgement'. If this changes the peer selection
+       -- governor will act on it
+       currentLedgerStateJudgement :: !LedgerStateJudgement,
+
+       -- | Has the governor fully reset its state
+       hasOnlyBootstrapPeers       :: !Bool
```

Comment:
Same as my previous reply. We need to cache this. These values will be updated by a monitoring action that is waiting on the STM actions. I believe this fits nicely with the current design of the governor

Reply:
The previous one we use quite extensively in various places, this might not be the case here, isn't it? If it's just used sporadically or in a few places then I'd avoid it and rather have a function which computes it on the fly.

_one of the hardest problem of CS is cache invalidation :wink:_

Reply:
This value is basically a flag that tells the governor whether the node has reset its state and only has trusted peers in its known peer set. It does not have cache semantics.

Reply:
which lines, it seems your link doesn't point to anything in particular